server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStats.java

```diff
@@ -115,6 +120,11 @@ public double value(String name) {
         return super.value(name);
     }
 
+    @Override
+    public Iterable<String> valueNames() {
+        return METRIC_NAMES;
```

Comment:
I'd feel better if we returned an immutable version or a copy here.  Or maybe just make `METRIC_NAMES` immutable to begin with.

Reply:
I had thought `toSet` was immutable - looks like it isn't. Yeah, I'd make `METRIC_NAMES` immutable. Its generally confusing to me when something in CONSTANT_CASE isn't actually immutable.

Reply:
:+1: I wrapped them all using `Collections.unmodifiableSet(...)`